### PR TITLE
Fix cost functions and raise test coverage

### PR DESCRIPTION
**Pull request by diego-urgell**
Fix norm_mean, poisson, and exponential distributions so that the add the missing values on R code. Also, make sure that the distributions return `-2*logLik`.

Raise the test coverage by testing the R interface.

### CODE REVIEW

**Comment by diego-urgell** [comment]
Hi @rkillick and @tdhock!

I have a question regarding the cost function for the Poisson distribution. This is my final expression, the part that says R is the one that is added on the R code, and the rest is implemented in C++. 

IMG_B33FB378DF2C-1

Is it correct? It is different from the expression used in the `changepoint` package (`logLik` functions for Poisson distribution), so I am not sure if it is correct to include all the terms. 

Also, on the exponential distribution, my cost equation is the following: 

IMG_007423883AD2-1

It is very similar to the expression on `changepoint` (also taking into account the `logLik` function), but the difference is that there the `T` is not added. 

I just want to confirm that the equations are correct in order to return the most appropriate cost value to the user, `-2*logLik`.

**Comment by tdhock** [comment]
you can use dpois to check.
typically I use mean - data * log(mean) which is the same up to a constant

**Comment by diego-urgell** [comment]
@tdhock and @rkillick I added several tests to the package. Among them, I created a function to compute the cost of the model in R code given the predicted changepoints, for every distribution. 

I realized I was not testing if the cost value was correct. Even tough I verified it using the `changepoint` package, I did not save the values and I thought it was better to implement the functions in R code. Everything works as expected 😄

**Comment by diego-urgell** [comment]
@tdhock @rkillick I finished adding more tests to the package. Now all the cost functions are tested. Also the resid method (for the implemented distributions is tested). 

This was of *great* importance because I found that there was a bug on the `meanvar_norm` and `var_norm` distributions, were I was calculating the variance of the right segment considering one unit less on the length, so the variance parameter was wrong. This is fixed now. 

Overall, there are 85 tests on the package and there will be a coverage of 98%. Tomorrow I will probably check whats missing to lift it to 100%! 

Also I will create the Readme with an explanation of the package, a few examples, and the work that was done during GSOC. The objective is to create a tag with a first release that conveys the work done during this time. I will also fix the R code that @tdhock suggested :)

File: tests/testthat/test-RInterface.R [hunk 1]
```diff
@@ -6,6 +6,55 @@
...
+
+  dist <- dist(object)
+  obj_coef <- coef(object, ncpts)
+  resids <- resid(object)
+  data <- object@data
+
+  if (dist == "mean_norm"){ # Cost function per segment.
+    resid_func <- function(start, end){
+      tmp <- resids[start:end] * sd(data)
+      tmp <- tmp + mean(data[start:end])
```
Comment (tdhock) [inline tests/testthat/test-RInterface.R]: @diego-urgell please use a more informative name than `tmp` if possible.
Also please avoid re-writing /changing the same variable `tmp <- tmp + something`
both are very confusing for others who read your code.

File: tests/testthat/test-RInterface.R [hunk 2]
```diff
@@ -6,6 +6,55 @@
...
+}
+
+check_resid <- function(object, ncpts=nrow(object@models_summary)){
+
+  dist <- dist(object)
+  obj_coef <- coef(object, ncpts)
+  resids <- resid(object)
+  data <- object@data
+
+  if (dist == "mean_norm"){ # Cost function per segment.
```
Comment (tdhock) [inline tests/testthat/test-RInterface.R]: here you can simplify the code by changing
```r
if(dist=="mean_norm")resid_func <- something
else if(dist=="var_norm")resid_func <- something_else
```
you can instead do
```r
resid_func <- if(dist=="mean_norm")something else if(dist=="var_norm") something_else
```

Comment (diego-urgell) [reply tests/testthat/test-RInterface.R]: Done! Thanks @tdhock :)

**Pull request merged**